src/DataStructures/DataVector.hpp

```diff
@@ -647,6 +677,24 @@ DataVector& DataVector::operator=(const blaze::Vector<VT, VF>& expression) {
   data_ = expression;
   return *this;
 }
+
+template <typename VT, bool VF>
+SPECTRE_ALWAYS_INLINE decltype(auto) dot_product(
+    const blaze::Vector<VT, VF>& lhs, const blaze::Vector<VT, VF>& rhs) {
+  return dot(lhs, rhs);
+}
+
+template <typename VT, bool VF>
+SPECTRE_ALWAYS_INLINE decltype(auto) magnitude_square(
+    const blaze::Vector<VT, VF>& expression) {
+  return dot(expression, expression);
```

Comment:
Blaze implements `sqrLength` (as used above in the `magnitude_square` function for a `DataVector`) only for `blaze::DenseVector` and `blaze::SparseVector` (see https://bitbucket.org/blaze-lib/blaze/wiki/Vector%20Operations#!length-sqrlength). Can someone clarify for me why that is? And what's the best way to wrap it?